src/neo/Ledger/Blockchain.cs

```diff
@@ -431,14 +431,19 @@ private void OnParallelVerified(ParallelVerified parallelVerified)
             RelayResultReason reason = parallelVerified.VerifyResult;
             if (reason == RelayResultReason.Succeed)
             {
-                if (View.ContainsTransaction(parallelVerified.Transaction.Hash))
-                    reason = RelayResultReason.AlreadyExists;
-                else if (!MemPool.CanTransactionFitInPool(parallelVerified.Transaction))
-                    reason = RelayResultReason.OutOfMemory;
-                else if (!MemPool.TryAdd(parallelVerified.Transaction.Hash, parallelVerified.Transaction))
-                    reason = RelayResultReason.OutOfMemory;
-                else if (parallelVerified.ShouldRelay)
-                    system.LocalNode.Tell(new LocalNode.RelayDirectly { Inventory = parallelVerified.Transaction });
+                reason = parallelVerified.Transaction.VerifyForEachBlock(currentSnapshot, MemPool.SendersFeeMonitor.GetSenderFee(parallelVerified.Transaction.Sender));
```

Comment:
Should we remove L410?

Reply:
I think no. We need to check the L410 line in advance to avoid wasting resources on parallel VerifyWitness for unqualified transactions.

Reply:
I am not sure, we save resource in certain cases and waste in other, we should check what it will be more usual.

Reply:
Perhaps reverting to single-threaded mode is currently the best option.

Reply:
This was this alternative https://github.com/neo-project/neo/pull/1429 but the speed it's lower